persistence-modules/spring-data-jpa-query-2/src/test/java/com/baeldung/persistence/hibernate/FooSortingPersistenceIntegrationTest.java

```diff
@@ -120,30 +124,45 @@ public final void whenHQlSortingByMultipleAttributes_andOrderDirection_thenPrint
 
     @Test
     public final void whenHQLCriteriaSortingByOneAttr_thenPrintSortedResults() {
-        final Criteria criteria = session.createCriteria(Foo.class, "FOO");
-        criteria.addOrder(Order.asc("id"));
-        final List<Foo> fooList = criteria.list();
+        CriteriaQuery<Foo> selectQuery = session.getCriteriaBuilder().createQuery(Foo.class);
+        selectQuery.from(Foo.class);
+        Query<Foo> query = session.createQuery(selectQuery);
+
+        query.setOrder(Collections.singletonList(Order.asc(Foo.class,"id")));
+        final List<Foo> fooList = query.list();
         for (final Foo foo : fooList) {
             LOGGER.debug("Id: {}, FirstName: {}", foo.getId(), foo.getName());
         }
     }
 
     @Test
     public final void whenHQLCriteriaSortingByMultipAttr_thenSortedResults() {
-        final Criteria criteria = session.createCriteria(Foo.class, "FOO");
-        criteria.addOrder(Order.asc("name"));
-        criteria.addOrder(Order.asc("id"));
-        final List<Foo> fooList = criteria.list();
+
+        CriteriaQuery<Foo> selectQuery = session.getCriteriaBuilder().createQuery(Foo.class);
+        selectQuery.from(Foo.class);
+        Query<Foo> query = session.createQuery(selectQuery);
+
+        List<Order<? super Foo>> orderBy = new ArrayList<>(2);
+        orderBy.add(Order.asc(Foo.class,"name"));
+        orderBy.add(Order.asc(Foo.class,"id"));
+        query.setOrder(orderBy);
+        final List<Foo> fooList = query.list();
         for (final Foo foo : fooList) {
             LOGGER.debug("Id: {}, FirstName: {}", foo.getId(), foo.getName());
         }
     }
 
     @Test
     public final void whenCriteriaSortingStringNullsLastAsc_thenNullsLast() {
-        final Criteria criteria = session.createCriteria(Foo.class, "FOO");
-        criteria.addOrder(Order.asc("name").nulls(NullPrecedence.LAST));
-        final List<Foo> fooList = criteria.list();
+        CriteriaQuery<Foo> selectQuery = session.getCriteriaBuilder().createQuery(Foo.class);
+        selectQuery.from(Foo.class);
+        Query<Foo> query = session.createQuery(selectQuery);
+
```

Comment:
Is adding initial capacity (2) needed?

Reply:
if we know beforehand that the initial capacity of the list is 2, it's better to specify it.

Reply:
but it's not needed it will work anyway.